bundles/AdminBundle/Controller/Admin/AssetController.php

```diff
@@ -120,42 +120,10 @@ public function getDataByIdAction(Request $request, EventDispatcherInterface $ev
                 $imageInfo['dimensions']['height'] = $asset->getHeight();
             }
 
-            $exifData = $asset->getEXIFData();
-            if (!empty($exifData)) {
-                $imageInfo['exif'] = $exifData;
-            }
-
-            $xmpData = $asset->getXMPData();
-            if (!empty($xmpData)) {
-                // flatten to a one-dimensional array
-                array_walk($xmpData, function (&$value) {
-                    if (is_array($value)) {
-                        $value = implode_recursive($value, ' | ');
-                    }
-                });
-                $imageInfo['xmp'] = $xmpData;
-            }
-
-            // check for VR meta-data
-            $mergedMetaData = array_merge($exifData, $xmpData);
-            if (isset($mergedMetaData['ProjectionType']) && $mergedMetaData['ProjectionType'] == 'equirectangular') {
-                $imageInfo['isVrImage'] = true;
-            }
-
-            $iptcData = $asset->getIPTCData();
-            if (!empty($iptcData)) {
-                // flatten data, to be displayed in grid
-                foreach ($iptcData as &$value) {
-                    if (is_array($value)) {
-                        $value = implode(', ', $value);
-                    }
-                }
-
-                $imageInfo['iptc'] = $iptcData;
-            }
-
             $imageInfo['exiftoolAvailable'] = (bool) \Pimcore\Tool\Console::getExecutable('exiftool');
 
+            $asset->handleEmbeddedMetaData(true);
```

Comment:
I think that can go, or replaced by `$asset->getEmbeddedMetaData();`

Reply:
And we can use the data from exiftool here as well if available, if not we should fall back to the former behavior, like that: 

if(!$asset->getEmbeddedMetaData()) {
   $asset->getEmbeddedMetaData(false); // read Exif, IPTC and XPM like in the old days ... 
}